Refactor sponsor data and introduce small logo sponsors on homepage

straight-shoota opened the pull request:
![grafik](https://github.com/crystal-lang/crystal-website/assets/466378/dac9c920-b4fc-40aa-bc0c-066d78e329cd)


Original preview:
![grafik](https://github.com/crystal-lang/crystal-website/assets/466378/512b67d1-5306-4e31-82c0-3e0c60eb872b)

beta-ziliani commented on scripts/merge.cr:
```diff
@@ -65,6 +65,8 @@ end
 all_sponsors.sort_by! { |s| {-s.last_payment, -s.all_time, s.since, s.name} }
 
 write_csv("sponsors.csv", all_sponsors)
+write_csv("sponsor_logos_l.csv", all_sponsors.select { |sponsor| sponsor.last_payment.to_i > 1000 })
```
This (might) need adjustment once we get the new levels.

beta-ziliani reviewed (commented):
This is good. We need to ensure the proper level, and get a confirmation with @jkicillof that the design on the home page is good with the new tier.

jkicillof commented:
I don't get the new feature, there're already sponsor logos at home page design, are you planing to add another hierarchy?
<img width="483" alt="image" src="https://github.com/crystal-lang/crystal-website/assets/5031636/a98db939-5a9f-44a4-a3cd-a7ed73ad648c">

jkicillof commented:
I didn't see the small logo in the preview 🤦‍♂️. I think we shouldn't have different logo sizes there. If we must do that I think we could add a badge to the best sponsors

straight-shoota commented:
Yeah we want to differentiate the sponsor logos based on their contribution value. A single class of sponsor logo throws everything in one basket and we want more discrete options to offer better perks for higher sponsorship tiers. This is part of an ongoing revamp of sponsorship tiers.

Differentiating on the basis of logo size is a common means to represent sponsorsship tiers. Size increments communicate a clear hiearchical order intuitively. I don't think a badge can do that efficiently.

ysbaddaden commented:
It's just the "works on arm" logo ain't the best logo for such a tiny size, unlike 84codes, Manas or Nikola ones, that would still fare well.

straight-shoota commented:
Yeah, perhaps we should increase the image size a bit to make it work better. We could consider skipping the text label then.

jkicillof commented:
I updated the sponsors section [here](https://projects.invisionapp.com/d/main#/console/20582111/432474609/preview). The logos on the second tier are smaller but not so much and they have less contrast.

straight-shoota commented:
I implemented these visuals keeping the same format but smaller logo size and lighter colors.

Getting the colors is a bit tricky, I'm using contrast and grayscale filters. It looks fine for the current set of logos.

Updated screenshot in the OP.

straight-shoota commented on scripts/merge.cr:
```diff
@@ -64,11 +64,15 @@ end
-  CSV.build(file) do |csv|
+write_csv("sponsors.csv", all_sponsors.select(&.listed?))
+write_csv("sponsor_logos_l.csv", all_sponsors.select { |sponsor| sponsor.last_payment.to_i > 1000 })
+write_csv("sponsor_logos_s.csv", all_sponsors.select { |sponsor| sponsor.last_payment.to_i.in?(350...1000) })
```
Actually, I think we should leave the old entry threshold for at least the rest of this month to keep the logo of PlaceOS around because they have already paid for it.

```suggestion
write_csv("sponsor_logos_s.csv", all_sponsors.select { |sponsor| sponsor.last_payment.to_i.in?(250...750) })
```

beta-ziliani commented on _data/others.json:
```diff
@@ -30,5 +30,14 @@
+    "url": "https://developer.arm.com/solutions/infrastructure/works-on-arm",
+    "logo": "sponsors/woa.png",
+    "last_payment": 500,
+    "all_time": 0,
```
For posterity, we're not listing the total, so there's no point in having this updated

beta-ziliani reviewed (approved):
Once the changes are committed, this is good to go
